src/pdstools/DAPHNEData.py

```diff
@@ -0,0 +1,256 @@
+"""
+DAPHNE data loading and storing class.
+"""
+
+# Internal Libraries
+import os
+
+# DUNE-DAQ Libraries
+import daqdataformats
+from hdf5libs import HDF5RawDataFile
+
+# defined here DUNE-DAQ/rawdatautils/pybindsrc/DAPHNEUnpacker.cpp
+
+from rawdatautils.unpack.daphne import (
+    np_array_adc,
+    np_array_channels,
+    np_array_timestamp,
+    get_n_frames
+)
+
+from rawdatautils.unpack.utils import *
+from fddetdataformats import *
+
+# External Libraries
+import numpy as np
+
+# pdstools
+from . import exceptions
+
+
+class DAPHNEData:
+    """
+    Class that loads a given raw HDF5 data file that contains
+    DAPHNE.
+
+    Loading fragments populates obj.ds_data. By default this only loads
+    a few fragments as the data rate for PDS is very high.
+
+    Gives warnings and information when trying to load empty fragments.
+    These outputs can be quieted with quiet=True on init.
+    """
+    # Useful print colors
+    _FAIL_TEXT_COLOR = '\033[91m'
+    _WARNING_TEXT_COLOR = '\033[93m'
+    _BOLD_TEXT = '\033[1m'
+    _RESET_TEXT_COLOR = '\033[0m'
+
+    # Data
+    ds_data = np.array([], dtype=np.uint16).reshape(0, 1)  # Will concatenate
+    channels = np.array([], dtype=np.uint8)  # Will populate
+    _data_is_empty = True
+
+#----------------------------------------------------------------            
+    def __init__(self, filename: str, quiet: bool = False) -> None:
+        """
+        Loads the given HDF5 file and initializes member data.
+
+        Parameters:
+            filename (str): File name to load from.
+            quiet (bool): Quiet debugging output.
+
+        Returns:
+            Nothing.
+        """
+        # File structure loading
+        self._h5_file = HDF5RawDataFile(os.path.expanduser(filename))
+        self._quiet = quiet
+        self._set_fragment_paths(self._h5_file.get_all_fragment_dataset_paths())
+
+        # File identification attributes
+        self.run_id = self._h5_file.get_int_attribute("run_number")
+        self.file_index = self._h5_file.get_int_attribute("file_index")
+        return None
+
+#----------------------------------------------------------------            
+    def __call__(self, frag_path: str) -> np.ndarray:
+        """
+        Loads a fragment path, same as self.load_fragment.
+
+        Parameters:
+            frag_path (str): Fragment path to load.
+
+        Returns:
+            np.ndarray: Array of ADC values found in the fragment.
+                        Also concatenates self.ds_data with these values.
+        """
+        return self.load_fragment(frag_path)
```

Comment:
The documentation for `load_fragment` needs to change, so the documentation here also needs to change.